### PR TITLE
Fix some build warnings

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
Fix build error introduced in #950 and build warnings involving redefining `bool`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/mqtt/mqtt_demo_basic_tls/Makefile [hunk 1]
```diff
@@ -1,7 +1,8 @@
 CC = gcc
 
 INCLUDE_DIRS = -I . \
-               -I ../../../libraries/standard/mqtt/include
+               -I ../../../libraries/standard/mqtt/include \
+               -I ../../logging-stack/
```
Comment (leegeth) [inline demos/mqtt/mqtt_demo_basic_tls/Makefile:5]: do we have to maintain the Makefiles as we have migrated to cmake?

File: demos/mqtt/mqtt_demo_plaintext/Makefile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 CC = gcc
 
 INCLUDE_DIRS = -I . \
-               -I ../../../libraries/standard/mqtt/include
+               -I ../../../libraries/standard/mqtt/include \
+               -I ../../logging-stack/
```
Comment (leegeth) [inline demos/mqtt/mqtt_demo_plaintext/Makefile:5]: do we have to maintain the Makefiles as we have migrated to cmake?

File: demos/mqtt/mqtt_demo_basic_tls/Makefile (hunk 1 above)
Comment (muneebahmed10) [reply demos/mqtt/mqtt_demo_basic_tls/Makefile:5]: Possibly not, but I don't think we're ready to remove them yet. If anyone's using a Mac then CMake won't work but the Makefile should

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 3]
```diff
@@ -23,9 +23,9 @@
 #define MQTT_LIGHTWEIGHT_H
 
 /* bools are only defined in C99+ */
-#if __STDC_VERSION__ >= 199901L
+#if defined( __cplusplus ) || __STDC_VERSION__ >= 199901L
```
Comment (aggarw13) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:26]: (Probably out-of-scope of the PR). Do we want to keep such a check in the code? I thought we are heading to keep no references of C99 in the code?

Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:26]: We either need the check or change all the booleans to some other macro or enum. There hasn't been any indication that we don't want to use the `bool` macro, so this defines it for C90.

Comment (aggarg) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:26]: > change all the booleans to some other macro or enum. 

May be better to do.

**Pull request merged**